### PR TITLE
Update the allocate_for_instance method to juno version

### PR DESCRIPTION
**Pull request by rodis**
Change-Id: Ifad3c1de00bbc95ed869b67aaf1a8d9336087dce
Signed-off-by: Rosario Di Somma <rosario.disomma@dreamhost.com>

### CODE REVIEW

File: akanda/nova/neutronv2/api.py [hunk 1]
```diff
@@ -61,55 +64,72 @@ def allocate_for_instance(self, context, instance, **kwargs):
             # pre-allocated port we also remove it from this set.
             available_macs = set(hypervisor_macs)
         neutron = neutronv2.get_client(context)
-        LOG.debug(_('allocate_for_instance() for %s'),
-                  instance['display_name'])
-        if not instance['project_id']:
+        LOG.debug('allocate_for_instance()', instance=instance)
```
Comment (agentultra) [inline akanda/nova/neutronv2/api.py]: Missing il8n here

Comment (markmcclain) [inline akanda/nova/neutronv2/api.py]: OpenStack stopped adding translation wrappers to debug messages during the Juno cycle.   I think it is a pattern we should continue to follow.

Comment (agentultra) [inline akanda/nova/neutronv2/api.py]: Ah, my bad. Forgot about that. Was just looking at the rest of the source. Maybe we should clear them all out in this patch then?

**Comment by markmcclain** [comment]
LGTM

File: akanda/nova/neutronv2/api.py (hunk 1 above)
Comment (rodis) [inline akanda/nova/neutronv2/api.py]: The remaining translation wrappers are used for LOG.info and LOG.warning which should be still in use in Openstack I believe.

Comment (agentultra) [inline akanda/nova/neutronv2/api.py]: Cool. Never mind me then. :+1:

File: akanda/nova/neutronv2/api.py [hunk 2]
```diff
@@ -275,32 +309,30 @@ def deallocate_for_instance(self, context, instance, **kwargs):
...
-                    with excutils.save_and_reraise_exception():
-                        LOG.exception(_("Failed to delete neutron port %s"),
-                                      port)
+            if port['device_owner'].startswith('network:'):
+                body = dict(device_id='')
+                neutron.update_port(port['id'], dict(port=body))
+            else:
+                ports_to_delete.append(port['id'])
+
+        self._delete_ports(neutron, instance, ports_to_delete,
```
Comment (ryanpetrello) [inline akanda/nova/neutronv2/api.py]: We're no longer doing any exception handling here; can you give any explanation (e.g., the exception handling moved into `_delete_ports`)?

**Pull request merged**